### PR TITLE
PPTP-1541 Changes to remove the trailing and leading spaces error mes…

### PR DESCRIPTION
**Pull request by kiranprakashdigital**
…sages.

### Description of Work carried through

Brief description of what/why/how.

#### Check list 
 - [ ] `./precheck` was executed (Integration/Component/Unit tests)
 - [ ] `sbt scalafmt test:scalafmt` was executed
 - [ ] Required Environment Config has been amended/added
 - [ ] Links to dependencies have been included (BE/FE work)
 - [ ] User Acceptance Tests (UAT) were run locally and have passed.
 - [ ] No Accessibility errors/warnings reported by Wave

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtax/registration/forms/liability/LiabilityStartDate.scala [hunk 1]
```diff
@@ -27,22 +27,15 @@ object LiabilityStartDate {
   val dateFormattingError = "liabilityStartDate.formatting.error"
   val dateOutOfRangeError = "liabilityStartDate.outOfRange.error"
 
-  val dateLowerLimit: LocalDate = LocalDate.of(2022, 4, 1)
-
-  private val dateUpperLimit =
-    if (LocalDate.now().isAfter(dateLowerLimit)) LocalDate.now().plusYears(1)
-    else dateLowerLimit.plusYears(1)
+  val dateLowerLimit: LocalDate = LocalDate.now()
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/forms/liability/LiabilityStartDate.scala:30]: Are you sure about this change? It looks like 01/04/2022 is the date on which the tax comes into law.

File: test/uk/gov/hmrc/plasticpackagingtax/registration/forms/liability/LiabilityStartDateSpec.scala [hunk 2]
```diff
@@ -55,7 +47,7 @@ class LiabilityStartDateSpec extends AnyWordSpec with Matchers {
 
       "is exactly one year after tax liability starts (01-04-2022)" in {
 
-        val date = dateLowerLimit.plusYears(1)
+        val date = dateLowerLimit.minusYears(1)
```
Comment (RyanKellyHMRC) [inline test/uk/gov/hmrc/plasticpackagingtax/registration/forms/liability/LiabilityStartDateSpec.scala:50]: I think this should be `plusYears(1)` in order to test what is suggested by the test name.

**Review by RyanKellyHMRC — COMMENTED** [review]
Couple of questions

File: app/uk/gov/hmrc/plasticpackagingtax/registration/forms/liability/LiabilityStartDate.scala (hunk 1 above)
Comment (kiranprakashdigital) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/forms/liability/LiabilityStartDate.scala]: I have changed it..

File: test/uk/gov/hmrc/plasticpackagingtax/registration/forms/liability/LiabilityStartDateSpec.scala (hunk 2 above)
Comment (kiranprakashdigital) [reply test/uk/gov/hmrc/plasticpackagingtax/registration/forms/liability/LiabilityStartDateSpec.scala]: Changed this as well

File: app/uk/gov/hmrc/plasticpackagingtax/registration/forms/liability/LiabilityStartDate.scala [hunk 3]
```diff
@@ -27,15 +27,20 @@ object LiabilityStartDate {
...
     Try(LocalDate.parse(date.toString)).isSuccess
 
   private val isDateInRange: Date => Boolean = date =>
     (LocalDate.parse(date.toString).isEqual(dateLowerLimit) || LocalDate.parse(
       date.toString
-    ).isBefore(dateLowerLimit))
+    ).isBefore(dateLowerLimit)) &&
+      (LocalDate.parse(date.toString).isEqual(dateUpperLimit) || LocalDate.parse(
+        date.toString
+      ).isAfter(dateUpperLimit))
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/forms/liability/LiabilityStartDate.scala:43]: Have you double checked this with Jon?

**Review by RyanKellyHMRC — APPROVED** [review]
Approved but just checking that you've checked the data validation with Jon.

File: conf/messages.en [hunk 4]
```diff
@@ -48,6 +48,7 @@ date.month.decimal.error = Month must not include decimals
 date.month.outOfRange.error = Month must be a number between 1 and 12
 date.month.leadingBlankSpace.error = Month must not have leading spaces
 date.month.trailingBlankSpace.error = Month must not have trailing spaces
+date.decimal.error = Date must be a real date
```
Comment (tmc-ee) [inline conf/messages.en:51]: Does this need changing/renaming from `date.decimal.error` as we're removing decimal specific error messages? If it's been left like this for easier mental processing/readability then ignore no worries :+1:

File: conf/messages.en [hunk 5]
```diff
@@ -112,7 +113,7 @@ liabilityStartDatePage.title = What is the date you first became liable?
 liabilityStartDatePage.hint = For example, 12 11 2007
 liabilityStartDatePage.question = What is the date you first became liable?
 liabilityStartDate.formatting.error = Liability start date - wrong format
-liabilityStartDate.outOfRange.error = Liability start date must be on or after 1 April 2022 and before 2 April 2023
+liabilityStartDate.outOfRange.error = Liability date must be today or in the past
```
Comment (tmc-ee) [inline conf/messages.en:116]: I know we discussed this in 3As but in retrospect should this be more along the lines of `Liability start date must be on or after 1 April 2022`? Stating `in the past` specifically could lead to a lot of "but 31/03/2022 is in the past?" type queries... I'm not sure, what do you think?

File: conf/messages.en (hunk 4 above)
Comment (kiranprakashdigital) [reply conf/messages.en:51]: We will revisit this once we get the actual error message.

File: conf/messages.en (hunk 5 above)
Comment (kiranprakashdigital) [reply conf/messages.en:116]: The error message in this context is to be decided by the content people.

**Review by tmc-ee — APPROVED** [review]

**Pull request merged**